
syntax = "proto3";

message DatasetIndex {
  // Dataset index version
  string version = 1;
  repeated DatasetEntry entries = 2;
}

message DatasetEntry {
  string id = 1;
  oneof metadata {
    MusicTrackMetadata music_metadata = 2;
  }
  map<string, EntryFile> files = 3;
  string split = 4;

  // -------------------------- ENTRY METADATA MESSAGES DEFINITIONS --------------------------
  message MusicTrackMetadata {
    string composer = 1;
    string title = 2;
    int32 year = 4;
    double duration = 5;
    string release = 6;
  }
}

message EntryFile {
  string path = 1;
  string md5_checksum = 2;
  oneof attributes {
    SymbolicMusicFileAttributes symbolic_music_attributes = 3;
  }

  // -------------------------- FILE ATTRIBUTES MESSAGES DEFINITIONS --------------------------
  message SymbolicMusicFileAttributes {
    double match_score = 1;
  }
}
